File: 2021/day01-1.hs
import System.IO

solution :: String -> Int
solution content = length . filter id . isIncreasing . map read . lines $ content

isIncreasing :: [Int] -> [Bool]
isIncreasing (x:y:[]) = [y > x]
isIncreasing (x:y:xs) = [y > x] ++ (isIncreasing ([y] ++ xs))

main :: IO ()
main = do
    handle <- openFile "day01.dat" ReadMode
    content <- hGetContents handle
    print . solution $ content

File: 2021/day01-1.ts
function solution(content: string) {
  const depths = content.split("\n").map((item) => parseInt(item));
  return depths
    .slice(1)
    .map((item, index) => item > depths[index])
    .filter((item) => item).length;
}

Deno.readTextFile("day01.dat")
  .then((content) => content.trim())
  .then(solution)
  .then(console.log);
